src/process_builder.rs

```diff
@@ -0,0 +1,271 @@
+use std::collections::HashMap;
+use std::env;
+use std::ffi::{OsStr, OsString};
+use std::fmt;
+use std::path::Path;
+use std::process::{Command, Output, Stdio};
+
+use failure::{Fail, format_err};
+use jobserver::Client;
+use shell_escape::escape;
+
+/// A builder object for an external process, similar to `std::process::Command`.
+#[derive(Clone, Debug)]
+pub struct ProcessBuilder {
```

Comment:
Borrowed this from cargo as it is a very nice cross-platform implementation of exec_replace. `std::process::Command` causes weird IO issues with http-prompt

Reply:
preliminary comment: I think it's worth linking to https://github.com/rust-lang/cargo/blob/master/src/cargo/util/process_builder.rs just so folks can see this source for themselves as well :)